mindsdb/utilities/otel.py

```diff
@@ -1,44 +1,152 @@
 import os
+import typing
 
-from opentelemetry import trace
+from opentelemetry import trace  # noqa: F401
+from opentelemetry.exporter.otlp.proto.grpc._log_exporter import OTLPLogExporter
+from opentelemetry.exporter.otlp.proto.grpc.metric_exporter import OTLPMetricExporter
+from opentelemetry.sdk.metrics._internal.export import ConsoleMetricExporter, MetricExporter
 from opentelemetry.sdk.resources import Resource
-from opentelemetry.sdk.trace import TracerProvider, Span
-from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
+from opentelemetry.sdk.trace.export import ConsoleSpanExporter
 from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
 
+from mindsdb.utilities.otel_logging import setup_otel_logging
+from mindsdb.utilities.otel_metrics import setup_otel_metrics
+from mindsdb.utilities.otel_tracing import setup_otel_tracing
+from mindsdb.utilities.utils import parse_csv_attributes
 from mindsdb.utilities import log
-logger = log.getLogger(__name__)
 
+logger = log.getLogger(__name__)
 
 # Check OpenTelemetry exporter type
 OTEL_EXPORTER_TYPE = os.getenv("OTEL_EXPORTER_TYPE", "console")  # console or otlp
 
 # Define OTLP endpoint. If not set, the default OTLP endpoint will be used
 OTEL_OTLP_ENDPOINT = os.getenv("OTEL_OTLP_ENDPOINT", "http://localhost:4317")
 
+# Define OTLP logging endpoint. If not set, the default OTLP logging endpoint will be used
+OTEL_OTLP_LOGGING_ENDPOINT = os.getenv("OTEL_OTLP_LOGGING_ENDPOINT", OTEL_OTLP_ENDPOINT)
+
+# Define OTLP tracing endpoint. If not set, the default OTLP tracing endpoint will be used
+OTEL_OTLP_TRACING_ENDPOINT = os.getenv("OTEL_OTLP_TRACING_ENDPOINT", OTEL_OTLP_ENDPOINT)
+
+# Define OTLP metrics endpoint. If not set, the default OTLP metrics endpoint will be used
+OTEL_OTLP_METRICS_ENDPOINT = os.getenv("OTEL_OTLP_METRICS_ENDPOINT", OTEL_OTLP_ENDPOINT)
+
 # Define service name
-OTEL_SERVICE_NAME = os.getenv("OTEL_SERVICE_NAME", "mindsdb")
+OTEL_SERVICE_NAME = os.getenv("OTEL_SERVICE_NAME", "mindsdb_new_test")
```

Comment:
Is this intentionally test text?

Reply:
Nope. It was from a test and I'm removing now